azurerm/resource_arm_sql_administrator_test.go

```diff
@@ -0,0 +1,184 @@
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMSqlAdministratorExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serverName := rs.Primary.Attributes["server_name"]
+
+		client := testAccProvider.Meta().(*ArmClient).sqlServerAzureADAdministratorsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		_, err := client.Get(ctx, resourceGroup, serverName, "activeDirectory")
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMSqlAdministratorDisappears(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serverName := rs.Primary.Attributes["server_name"]
+
+		client := testAccProvider.Meta().(*ArmClient).sqlServerAzureADAdministratorsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		_, err := client.Delete(ctx, resourceGroup, serverName, "activeDirectory")
+		if err != nil {
+			return fmt.Errorf("Bad: Delete on sqlAdministratorClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMSqlAdministratorDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_sql_administrator" {
+			continue
+		}
+
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serverName := rs.Primary.Attributes["server_name"]
+
+		client := testAccProvider.Meta().(*ArmClient).sqlServerAzureADAdministratorsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := client.Get(ctx, resourceGroup, serverName, "activeDirectory")
+		if err != nil {
+			return err
+		}
+
+		return fmt.Errorf("SQL Administrator (server %q / resource group %q) still exists: %+v", serverName, resourceGroup, resp)
+	}
+
+	return nil
+}
+
+func testAccAzureRMSqlAdministrator_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+data "azurerm_client_config" "current" {}
+
+resource "azure_resource_group" "test" {
+	name = "acctestRG_%d"
+	location = "%s"
+}
+
+resource "azurerm_sql_server" "test" {
+	name = "acctestsqlserver%d"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    location = "${azurerm_resource_group.test.location}"
+    version = "12.0"
+    administrator_login = "mradministrator"
+    administrator_login_password = "thisIsDog11"
+}
+
+resource "azurerm_sql_administrator" "test" {
+	server_name = "${azurerm_sql_server.test.name}"
+	resource_group_name = "${azurerm_resource_group.test.name}"
+	login = "sqladmin"
+    tenant_id = "${data.azurerm_client_config.current.tenant_id}"
+    object_id = "${data.azurerm_client_config.current.client_id}"
+}
```

Comment:
_minor_ could we make the spacing consistent?

Reply:
fixed